### ISSUE TITLE
Azure Quickstart Templates for ASA to be provided

### ISSUE THREAD

**Issue by ezYakaEagle442** [Azure/Azure-Spring-Apps#28]
**Is your feature request related to a problem? Please describe.**
There is no ARM/Nicep Template for ASA.
The only one is outdated since 2 years at [https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.appplatform/azure-spring-cloud](https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.appplatform/azure-spring-cloud)

**Describe the solution you'd like**
At least 2 FULL Bicep Template + ARM Template should be provided as quick starts :

1. minimum settings with Standard Tier
2. ALL settings with Enterprise Tier including SSO, Cloud Gateway, API Portal, Build Service, Dev tools/ACME Accelerator

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by allxiao** [comment]
Currently, we leverage the automated generated API and Bicep documentation from our API swagger definitions. You can check the documentation site for reference.

* [Microsoft.AppPlatform/Spring - Bicep, ARM template & Terraform AzAPI reference | Microsoft Learn](https://learn.microsoft.com/en-us/azure/templates/microsoft.appplatform/spring?pivots=deployment-language-bicep)

The docs provide full templates for all the Azure Spring Apps resources. If you want to get the effective part of the operations mentioned in the Azure Spring Apps documentation and written in Azure CLI commands, you can add `--debug --verbose` to the given command and check the payload passed to ARM. You can modify the Bicep / ARM template accordingly.

We put the task to add Bicep code samples in public documentation into Backlog, while we will actively improve the description for each API field to make it more clear what this field is and the values it will accept.

**Comment by ezYakaEagle442** [comment]
> Currently, we leverage the automated generated API and Bicep documentation from our API swagger definitions. You can check the documentation site for reference.
> 
> * [Microsoft.AppPlatform/Spring - Bicep, ARM template & Terraform AzAPI reference | Microsoft Learn](https://learn.microsoft.com/en-us/azure/templates/microsoft.appplatform/spring?pivots=deployment-language-bicep)
> 
> The docs provide full templates for all the Azure Spring Apps resources. If you want to get the effective part of the operations mentioned in the Azure Spring Apps documentation and written in Azure CLI commands, you can add `--debug --verbose` to the given command and check the payload passed to ARM. You can modify the Bicep / ARM template accordingly.
> 
> We put the task to add Bicep code samples in public documentation into Backlog, while we will actively improve the description for each API field to make it more clear what this field is and the values it will accept.

NO, this is just the Bicep template doc for ASA, it does NOT help, anyone can find this doc, but without real values then it is useless

The ask here is to provide actual values in a snippet like I did at 
[https://github.com/ezYakaEagle442/azure-spring-apps-petclinic-mic-srv/blob/azure/iac/bicep/modules/asa/asa-e.bicep](https://github.com/ezYakaEagle442/azure-spring-apps-petclinic-mic-srv/blob/azure/iac/bicep/modules/asa/asa-e.bicep)

[https://github.com/ezYakaEagle442/azure-spring-apps-petclinic-mic-srv/blob/azure/iac/bicep/modules/asa/asa-e-api-portal-gateway.bicep](https://github.com/ezYakaEagle442/azure-spring-apps-petclinic-mic-srv/blob/azure/iac/bicep/modules/asa/asa-e-api-portal-gateway.bicep)

**Comment by hangwan97** [comment]
In the past several months, the structure of the doc has been updated after the azure spring cloud to azure spring apps name change.

The functions and usages of Quickstart have also changed accordingly as the main purpose of quickstart is just to provide day 1 quick and easy walk-through.

The team is now adding quickstart templates to the repo. Preview link: https://review.learn.microsoft.com/en-us/azure/spring-apps/quickstart-template?branch=pr-en-us-237947&pivots=sc-enterprise&tabs=Azure-portal.

As the quickstart is now extending to a list of popular spring workloads: https://learn.microsoft.com/en-us/azure/spring-apps/quickstart-deploy-event-driven-app-standard-consumption?pivots=sc-standard
web app
event-driven app
micro-service app(s)
...

The Bicep Template + ARM Template related docs are currently under the 'how-to': https://learn.microsoft.com/en-us/azure/spring-apps/quickstart-deploy-infrastructure-vnet?tabs=azure-spring-apps-standard%2Cazure-cli

The update of how-to will be updated in the upcoming months.

**Comment by moarychan** [comment]
The quickstart template for microservices(PetClinic) is available, please check [microservice-apps-enterprise-plan](https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.appplatform/microservice-apps-enterprise-plan).
